File: 03 - Largest Prime Factor/python/main.py
import math

def largestPrimeFactor(num):
    factor = 3
    previous = 1
    while factor < int(math.sqrt(num))+1:
        if num % factor == 0:
            previous = factor
            while num % factor == 0:
                num /= factor
        factor += 2

    return previous

result = largestPrimeFactor(600851475143)
print("result: " + str(result))

File: 03 - Largest Prime Factor/nim/main.nim
proc largest_prime_factor(input: int64): int64 =
    var num: int64 = input
    var factor: int64 = 3
    var previous: int64 = 3
    while factor <= num:
        if num mod factor == 0:
            previous = factor
            while num mod factor == 0:
                num = num div factor
        if factor * factor > input:
            break
        factor += 1
    previous

let result: int64 = largest_prime_factor(600851475143)
echo "result: ", result
